From cff2ffaa75a02aa19731ce4340b7e563b8595627 Mon Sep 17 00:00:00 2001
From: Sarika Sharma <quic_sarishar@quicinc.com>
Date: Mon, 24 Jun 2024 12:15:19 +0530
Subject: [PATCH] wifi :ath12k: Fix KW issue tx_stats size 4 may use index
 value upto 8

Currently size of array
'tx_stats->stats[ATH12K_STATS_TYPE_##name].transmit_type[0]' and
'tx_stats->stats[ATH12K_STATS_TYPE_##name].transmit_type[1]'
allocated is HAL_RX_RECEPTION_TYPE_MAX(4).
But while assigning the value for these array, the index used is
ppdu_type and ppdu_type enum max size is HTT_PPDU_STATS_PPDU_TYPE_MAX(8).

Here we are accessing the array more than its size that will
cause array outbound issue.

Hence, fix the issue by defining the size of array to
HTT_PPDU_STATS_PPDU_TYPE_MAX(8).

patch-dependency: 504-ath12k-add-debugfs-support.patch

Signed-off-by: Sarika Sharma <quic_sarishar@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/core.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/core.h b/drivers/net/wireless/ath/ath12k/core.h
index c9ab53e..318bab2 100644
--- a/drivers/net/wireless/ath/ath12k/core.h
+++ b/drivers/net/wireless/ath/ath12k/core.h
@@ -638,7 +638,7 @@ struct ath12k_htt_data_stats {
 	u64 bw[ATH12K_COUNTER_TYPE_MAX][ATH12K_BW_NUM];
 	u64 nss[ATH12K_COUNTER_TYPE_MAX][ATH12K_NSS_NUM];
 	u64 gi[ATH12K_COUNTER_TYPE_MAX][ATH12K_GI_NUM];
-	u64 transmit_type[ATH12K_COUNTER_TYPE_MAX][HAL_RX_RECEPTION_TYPE_MAX];
+	u64 transmit_type[ATH12K_COUNTER_TYPE_MAX][HTT_PPDU_STATS_PPDU_TYPE_MAX];
 	u64 ru_loc[ATH12K_COUNTER_TYPE_MAX][HAL_RX_RU_ALLOC_TYPE_MAX];
 };
 
-- 
2.34.1

